src/main/java/com/hpe/cloudfoundryjenkins/CloudFoundryPushPublisher.java

```diff
@@ -337,42 +337,78 @@ private boolean processOneApp(CloudFoundryClient client, DeploymentInfo deployme
+        String appTempName = deploymentInfo.getAppName();
+        String appTempURI = appURI;
+
+        if (cloudApplication != null) {
+           if (cutoverMethod.equals("redeploy")) {
+                 listener.getLogger().println("App already exists, redeploying.");
+                 client.deleteApplication(deploymentInfo.getAppName());
+                 listener.getLogger().println("App deleted.");
+           }
+           else if (cutoverMethod.equals("reroute")) {
+              // Alter app name and set temporary route to new app (blue/green)
+              listener.getLogger().println("App already exists, redeploying blue/green");
+              if (appTempName.endsWith("-green")) {
+                 appTempName = appTempName.replace("-green","-blue");
+                 appTempURI = appTempURI.replace("-green","-blue");
+              }
+              else if (appTempName.endsWith("-blue")) {
+                 appTempName = appTempName.replace("-blue","-green");
+                 appTempURI = appTempURI.replace("-blue","-green");
+              }
+              else {
+                 appTempName += "-green";
+                 appTempURI += "-green";
+              }
+           }
+           else {
+              listener.getLogger().println("App already exists, skipping creation.");
+              return false;
+           }
         }
 
-        // Create app if it doesn't exist
-        if (createNewApp) {
-            listener.getLogger().println("Creating new app.");
-            String stack = deploymentInfo.getStack();
-            if (client.getStack(stack) == null) {
-                throw new IllegalArgumentException("Stack " + stack + " does not exist on the target.");
-            }
-            Staging staging = new Staging(deploymentInfo.getCommand(), deploymentInfo.getBuildpack(),
-                    deploymentInfo.getStack(), deploymentInfo.getTimeout());
-            List<String> uris = new ArrayList<String>();
-            // Pass an empty List as the uri list if no-route is set
-            if (!deploymentInfo.isNoRoute()) {
-                uris.add(appURI);
-            }
-            List<String> services = deploymentInfo.getServicesNames();
-            client.createApplication(deploymentInfo.getAppName(), staging, deploymentInfo.getMemory(), uris, services);
+        // Create new app
+        Staging staging = new Staging(deploymentInfo.getCommand(), deploymentInfo.getBuildpack(),
+              null, deploymentInfo.getTimeout());
+        List<String> uris = new ArrayList<String>();
+        // Pass an empty List as the uri list if no-route is set
+        if (!deploymentInfo.isNoRoute()) {
+           uris.add(appTempURI);
         }
-
-        return createNewApp;
+        List<String> services = deploymentInfo.getServicesNames();
+        client.createApplication(appTempName, staging, deploymentInfo.getMemory(), uris, services);
```

Comment:
I didn't actually run this code since the 404 exception blocked me, but at first glance it looks like the new app will be created every time, even when the cutover method is "deploy".